### ISSUE TITLE
Unexpected value 'TreeModule' imported by the module 'X'

### ISSUE THREAD

**Issue by Gillardo** [500tech/angular2-tree-component#53]
I have downloaded the code from the master branch, built the code and tried to then use it in my application....

Now for some reason i get the error above? 

In my module, i am doing this
```
import { NgModule }           from '@angular/core';
import { CommonModule }       from '@angular/common';
import { FormsModule }        from '@angular/forms';
import { SharedModule } from '../shared/shared.module';
import { TreeModule } from 'angular2-tree-component/dist/angular2-tree-component';

import { PageComponent }   from './page.component';

import { routing } from './page.routing';

@NgModule({
    imports:        [ CommonModule, FormsModule, routing, SharedModule, TreeModule ],
    declarations:   [ PageComponent ]
})

export class PageModule {
}
```

I dont see anything wrong with what i have done??  Can someone help at all.

I have tried this with angular rc6 and rc7

**Comment by isaacplmann** [comment]
That code looks like it should work.  There's probably something going wrong with building the code.  The only difference in my app is that I do:
`import { TreeModule } from 'angular2-tree-component';`

But I'm pretty sure what you have should work too.  Double check that there is actually a TreeModule in the file.

It may also have to do with webpack somehow.  I haven't run into this myself, but @shlomiassaf mentioned he was running into [this error](https://github.com/angular/angular/issues/11438) with the 1.2.0 changes.

Can you run the webpack example included in angular2-tree-component and make sure that works for you?

**Comment by moff** [comment]
@Gillardo I have similar problem while using Webpack 2. Not an issue when I use Webpack 1.

**Comment by isaacplmann** [comment]
I'm not sure what I'm doing differently, but I'm using webpack 2 and its working.  I'm still using awesome-typescript-loader 1.1.1 though, maybe that is the issue.  I had problems (not this specifically) when upgrading to awesome-typescript-loader 2.x.

**Comment by Avien** [comment]
I'm also using awesome-typescript-loader 1.1.1 with Webpack 1 and it doesnt work

**Comment by Avien** [comment]
It seems its a problem casued by trying to import a NgModule from another node dependecey
(just like you did here with ```import { TreeModule } from 'angular2-tree-component/dist/angular2-tree-component';```
When I copy the module locally to my source the import works fine and no exception

**Comment by Avien** [comment]
Looks like npm link caused the problem
When I install the repo I am reting to import via npm git+ssh it works!
The problem now is that the workflow is harder without npm link

**Comment by Gillardo** [comment]
So basically, we need the new version on npm i guess?

**Comment by tscislo** [comment]
I can still see 1.1.18 version on npm. Can you please publish the new version?

**Comment by adamkleingit** [comment]
Published 1.2.0 to npm, please check

**Comment by AngularTx** [comment]
i have this same issue . I have reported the same . Some one let me know how this is fixed? 

"
It seems its a problem caused by trying to import a NgModule from another node dependency
(just like you did here with import { TreeModule } from 'angular2-tree-component/dist/angular2-tree-component';"

this is what is happening to me also. 

#276 is the ticket number

**Comment by adamkleingit** [comment]
@AngularTx This was fixed in 1.2.0, it has nothing to do with your issue
